### PR TITLE
LibWeb: added HTMLLinkElement.as

### PR DESCRIPTION
**Pull request by Hexeption**

### CODE REVIEW

File: Userland/Libraries/LibWeb/HTML/HTMLLinkElement.cpp [hunk 1]
```diff
@@ -90,6 +90,23 @@ void HTMLLinkElement::inserted()
     }
 }
 
+String HTMLLinkElement::as() const
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibWeb/HTML/HTMLLinkElement.cpp:93]: Hi, can you add a spec link above this?

File: Userland/Libraries/LibWeb/HTML/HTMLLinkElement.idl [hunk 2]
```diff
@@ -10,7 +10,7 @@ interface HTMLLinkElement : HTMLElement {
     [CEReactions, Reflect] attribute DOMString href;
     [FIXME, CEReactions] attribute DOMString? crossOrigin;
     [CEReactions, Reflect] attribute DOMString rel;
-    [FIXME, CEReactions] attribute DOMString as;
+    [CEReactions, Reflect] attribute DOMString as;
```
Comment (awesomekling) [inline Userland/Libraries/LibWeb/HTML/HTMLLinkElement.idl:13]: Hmm, doesn't using `Reflect` mean that your `as()` implementation won't even get called? (rather we just access the attribute directly)

Please add a test that confirms the expected behaviors :)

Comment (Hexeption) [reply Userland/Libraries/LibWeb/HTML/HTMLLinkElement.idl]: that was correct mb should be fixed now

File: Tests/LibWeb/Text/input/link_as.html [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
+<script src="include.js"></script>
+<script>
+    test(() => {
+        const link = document.createElement("link");
+        link.href = "https://fonts.googleapis.com/css?family=Roboto:400,700"
```
Comment (awesomekling) [inline Tests/LibWeb/Text/input/link_as.html:6]: Let's not trigger external HTTPS loads during tests.

File: Base/res/html/misc/linkas.html [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    link.as = 'style';
+
+    console.log("Link.as: " + link.as); // should be 'style'
+
+    link.as = 'uwu';
+
+    console.log("Link.as: " + link.as); // should be ''
+</script>
+</body>
+</html>
```
Comment (awesomekling) [inline Base/res/html/misc/linkas.html:21]: I don't think we need this file since we already have an automatic test.

**Review by awesomekling — CHANGES_REQUESTED** [review]
Thanks for working on this :)

**Comment by awesomekling** [comment]
Next time, please don't add a new file just to remove it in the next commit. :)

File: Userland/Libraries/LibWeb/HTML/HTMLLinkElement.cpp [hunk 5]
```diff
@@ -90,6 +90,29 @@ void HTMLLinkElement::inserted()
     }
 }
 
+// https://html.spec.whatwg.org/multipage/semantics.html#dom-link-as
+String HTMLLinkElement::as() const
+{
+    String attribute_value = get_attribute_value(HTML::AttributeNames::as);
+
+    if (attribute_value.equals_ignoring_ascii_case("fetch"sv) ||
```
Comment (nico) [inline Userland/Libraries/LibWeb/HTML/HTMLLinkElement.cpp:98]: fwiw, clang-format wants the `||`s at the start of the line instead.

Comment (nico) [reply Userland/Libraries/LibWeb/HTML/HTMLLinkElement.cpp:98]: Apologies, this was already fixed in #184.

**Pull request merged**